okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.idn
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isGreaterThan
+import okio.Buffer
+import okio.FileSystem
+import okio.Path.Companion.toPath
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class IdnaMappingTableTest {
+  private lateinit var table: IdnaMappingTable
+
+  @BeforeEach
+  fun setUp() {
+    table = FileSystem.RESOURCES.read("/okhttp3/internal/idna/IdnaMappingTable.txt".toPath()) {
+      readPlainTextIdnaMappingTable()
+    }
+  }
+
+  @Test fun regularMappings() {
+    assertThat("hello".map()).isEqualTo("hello")
+    assertThat("hello-world".map()).isEqualTo("hello-world")
+    assertThat("HELLO".map()).isEqualTo("hello")
+    assertThat("Hello".map()).isEqualTo("hello")
+    assertThat("¼".map()).isEqualTo("1⁄4")
```

Comment:
Maybe worth a few comments. Is the former a unicode character, and the latter a combined unicode sequence?

Reply:
Added!